File: MY PROGRAMS/LB/CPP/SELECTION,SEQUENCE AND  ITERATION/a4q1.cpp
/* Design application for hotel. According to the management team of hotel they are giving discount on total bill amount of customer. If bill bill amount is less than 500 then there is no discount , if bill amount is less than 1500 and more than 500 they give 10% discount. And if the bill amount is more than 1500 then they give 15% discount. Our application should accept total bill amount and depends on the discount policy we have to return the amount after applying discount.

Input:	1200
Output:	1080

Input:	290
Output:	290

Input:	3700
Output:	3145

*/




#include<iostream>

using namespace std;

class Bill
{
	public:
		float itotal=0.0,idis=0.0;

	public:
		float Return(float);
};

float Bill::Return(float iamt)
{
	if(iamt<500)
	{
		cout<<"There is no discount.";
	}
	else if((iamt>=500) && (iamt<1500))
	{
		idis=(iamt*10)/100;
		itotal=iamt-idis;
	}
	else if(iamt>=1500)
	{
		idis=(iamt*15)/100;
		itotal=iamt-idis;
	}

	return(itotal);
}

int main()
{
	float iret=0.0,ivalue=0.0;

	cout<<"Enter the bill ampount.";
	cin>>ivalue;

	Bill bobj;

	iret=bobj.Return(ivalue);

	cout<<"Total discount is Rs. "<<iret;

	return(0);
}

File: MY PROGRAMS/LB/JAVA/NUMBER/a4q1.java
/*	1. Design application for hotel. According to the management team of hotel they are giving discount on total bill amount of customer. 		If bill bill amount is less than 500 then there is no discount , if bill amount is less than 1500 and more than 500 they give 10% 		discount. And if the bill amount is more than 1500 then they give 15% discount. Our application should accept total bill amount and 		depends on the discount policy we have to return the amount after applying discount.

	Input:	1200
	Output:	1080

	Input:	3700
	Output:	3145

	Input:	290
	Output:	290	*/




import java.util.*;

class Assignment4
{
	public double Return(double damt)
	{
		double dtot=0.0,ddis=0.0;

		
		if(damt<500)
		{
			System.out.println("There is no discount.");

			return(0);
		}
		
		if((damt>=500) && (damt<1500))
		{
			dtot=(damt*10)/100;
		}
		else if(damt>=1500)
		{
			dtot=(damt*15)/100;
		}

		ddis=damt-dtot;

		return(ddis);
	}
}

class a4q1
{
	public static void main(String arg[])
	{
		Scanner sobj=new Scanner(System.in);

		double dvalue=0.0,dret=0.0;

		System.out.println("Enter the amount: ");
		dvalue=sobj.nextDouble();

		Assignment4 aobj=new Assignment4();

		dret=aobj.Return(dvalue);

		System.out.println("Discount is "+dret);
	}
}
